### PR TITLE
Update to sst-elements

### PR DESCRIPTION
**Pull request by sknigh**
- sst-elements: add dependencies, tweak build flags
- add dramsim3
- add dtc (dependency for riscv-isa-sim which may be added later)
- add flashdimmsim

### CODE REVIEW

**Comment by sknigh** [comment]
~@ reviewers, I made `check_compatibility` function in `sst-elements` package to throw errors if the user tries to use a variant is not yet supported. Is there a better way to do this?~

File: var/spack/repos/builtin/packages/dtc/package.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Dtc(MakefilePackage):
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/dtc/package.py:9]: (no changes required) FYI I'm fine with adding new packages to a PR if they are mentioned in some way by a dependency. In this case though this package appears to be totally separate: not mentioned by any of the other packages (and likewise this doesn't mention any of the other packages in this PR); normally I think such a package should be added in a separate PR.

File: var/spack/repos/builtin/packages/sst-elements/package.py [hunk 2]
```diff
@@ -69,35 +95,52 @@ def autoreconf(self, spec, prefix):
...
+            env['CC'] = spec['mpi'].mpicc
+            env['CXX'] = spec['mpi'].mpicxx
+            env['F77'] = spec['mpi'].mpif77
+            env['FC'] = spec['mpi'].mpifc
+
+        if "+pin" in spec:
+            args.append("--with-pin=%s" % spec["intel-pin"].prefix)
+
+        if "+dramsim2" in spec or "+hybridsim" in spec:
+            args.append("--with-dramsim=%s" % spec["dramsim2"].prefix)
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/sst-elements/package.py:114]: Was it intentional to replace 

```
args.append("--with-hybridsim=%s" % self.spec["hybridsim"].prefix)
```

with this? I still see a `hybridsim` dependency.

**Review by scheibelp — COMMENTED** [review]
I had a question:

File: var/spack/repos/builtin/packages/sst-elements/package.py (hunk 2 above)
Comment (sknigh) [reply var/spack/repos/builtin/packages/sst-elements/package.py:114]: This section's diff is a bit confusing. Most of the change is the result of replacing `self.spec` with a `spec` local variable (see L98) but the tool paired dissimilar lines. L110-111 is the updated version of L82-83. The diff matched lines L82-L83 with a `+dramsim3` dependency, which is new in this PR.

File: var/spack/repos/builtin/packages/dtc/package.py (hunk 1 above)
Comment (sknigh) [reply var/spack/repos/builtin/packages/dtc/package.py:9]: The commit message includes this line.
`- add dtc (dependency for riscv-isa-sim which may be added later)`

Since the intermediate package is missing there is no dependence in this PR.

File: var/spack/repos/builtin/packages/sst-elements/package.py (hunk 2 above)
Comment (scheibelp) [reply var/spack/repos/builtin/packages/sst-elements/package.py:114]: Ah ok should this add `conflicts(~hybridsim, when=~dramsim2)`?

Comment (sknigh) [reply var/spack/repos/builtin/packages/sst-elements/package.py:114]: Added conflicts for hybridsim

**Review by scheibelp — APPROVED** [review]

**Comment by scheibelp** [comment]
Thanks!
